driver-core/src/main/com/mongodb/operation/MixedBulkWriteOperation.java

```diff
@@ -481,6 +498,9 @@ private void addBatchResult(final BsonDocument result, final AsyncWriteBinding b
             executeBatchesAsync(binding, connection, nextBatch, retryWrites, callback);
         } else {
             if (batch.hasErrors()) {
+                if (retryWrites) {
+                    logUnableToRetry(batch.getPayload().getPayloadType().toString(), batch.getError());
```

Comment:
I have a feeling this isn't needed as it would be caught by line 464 - so could it be double logging here?